flair/training_utils.py

```diff
@@ -294,6 +297,172 @@ def _init_weights_index(self, key, state_dict, weights_to_watch):
+        min_lr (float or list): A scalar or a list of scalars. A
+            lower bound on the learning rate of all param groups
+            or each group respectively. Default: 0.
+        eps (float): Minimal decay applied to lr. If the difference
+            between new and old lr is smaller than eps, the update is
+            ignored. Default: 1e-8.
+
+    Example:
+        >>> optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
+        >>> scheduler = ReduceLROnPlateau(optimizer, 'min')
+        >>> for epoch in range(10):
+        >>>     train(...)
+        >>>     val_loss = validate(...)
+        >>>     # Note that step should be called after validate()
+        >>>     scheduler.step(val_loss)
+    """
+
+    def __init__(self, optimizer, mode='min', aux_mode='min', factor=0.1, patience=10,
+                 verbose=False, cooldown=0, min_lr=0, eps=1e-8):
+
+        if factor >= 1.0:
+            raise ValueError('Factor should be < 1.0.')
+        self.factor = factor
+
+        # Attach optimizer
+        if not isinstance(optimizer, Optimizer):
+            raise TypeError('{} is not an Optimizer'.format(
+                type(optimizer).__name__))
+        self.optimizer = optimizer
+
+        if isinstance(min_lr, list) or isinstance(min_lr, tuple):
+            if len(min_lr) != len(optimizer.param_groups):
+                raise ValueError("expected {} min_lrs, got {}".format(
+                    len(optimizer.param_groups), len(min_lr)))
+            self.min_lrs = list(min_lr)
+        else:
+            self.min_lrs = [min_lr] * len(optimizer.param_groups)
+
+        self.patience = patience
+        self.verbose = verbose
+        self.cooldown = cooldown
+        self.cooldown_counter = 0
+        self.mode = mode
+        self.aux_mode = aux_mode
+        self.best = None
+        self.best_aux = None
+        self.num_bad_epochs = None
+        self.mode_worse = None  # the worse value for the chosen mode
+        self.eps = eps
+        self.last_epoch = 0
+        self._init_is_better(mode=mode)
+        self._reset()
+
+    def _reset(self):
+        """Resets num_bad_epochs counter and cooldown counter."""
+        self.best = self.mode_worse
+        self.cooldown_counter = 0
+        self.num_bad_epochs = 0
+
+    def step(self, metric, auxiliary_metric = None):
+        # convert `metrics` to float, in case it's a zero-dim Tensor
+        current = float(metric)
+        epoch = self.last_epoch + 1
+        self.last_epoch = epoch
+
+        is_better = False
+
+        if self.mode == 'min':
+            if current < self.best:
+                is_better = True
+
+        if self.mode == 'max':
+            if current > self.best:
+                is_better = True
+
+        if current == self.best and auxiliary_metric:
+            current_aux = float(auxiliary_metric)
+            if self.aux_mode == 'min':
+                if current_aux < self.best_aux:
```

Comment:
Here 1self.best_aux1 has not been casted to float, but I do not know if it is necessary.

Reply:
You're right - it's currently not necessary but consistency it would be better to cast as well.